packages/manager/apps/container/src/payment-modal/PaymentModal.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState, useEffect } from 'react';
+import { Modal } from 'react-bootstrap';
+import { useTranslation } from 'react-i18next';
+
+import { useShell } from '@/context';
+
+import { fetchIcebergV6 } from '@ovh-ux/manager-core-api';
+
+import {
+  NO_DEFAULT_CARD,
+  EXPIRED_CARD,
+  SOON_EXPIRED_CARD
+} from './constants';
+
+import './styles.scss';
+
+interface IPaymentMethod {
+  icon?: any;
+  label: string;
+  status: string;
+  default: boolean;
+  oneclick: boolean;
+  lastUpdate: string;
+  description: string;
+  integration: string;
+  paymentType: string;
+  creationDate: string;
+  paymentMeanId: number;
+  expirationDate?: string;
+  paymentSubType: string;
+  paymentMethodId: number;
+}
+
+interface IPaymentMethodResponse {
+  data: IPaymentMethod[];
+}
+
+const PaymentModal = (): JSX.Element => {
+  const [mode, setMode] = useState('');
+  const { t } = useTranslation('payment-modal');
+  const [showPaymentModal, setShowPaymentModal] = useState(false);
+  const shell = useShell();
+
+  const paymentMethodURL = shell
+    .getPlugin('navigation')
+    .getURL('dedicated', '#/billing/payment/method');
+
+  useEffect(() => {
+    const getPaymentMethods = async () => {
+      const paymentResponse: IPaymentMethodResponse = await fetchIcebergV6({ route: '/me/payment/method' });
+      const hasDefaultPaymentMethod = paymentResponse.data && paymentResponse.data.length > 0 && paymentResponse.data.find(currentPaymentMethod => currentPaymentMethod.default);
+      let creditCardExpirationDate: Date;
+      let currentDateMinus30Days: Date;
+      let hasExpiredCreditCard = false;
+      let isSoonToBeExpireCreditCard = false;
+      const currentCreditCard = paymentResponse.data && paymentResponse.data.length > 0 && paymentResponse.data.find(currentPaymentMethod => currentPaymentMethod.paymentType === 'CREDIT_CARD');
+      if (currentCreditCard) {
+        creditCardExpirationDate = new Date(currentCreditCard.expirationDate);
+        currentDateMinus30Days = new Date();
+        currentDateMinus30Days.setDate(currentDateMinus30Days.getDate() - 30);
+        hasExpiredCreditCard = creditCardExpirationDate.getTime() < Date.now();
+        isSoonToBeExpireCreditCard = currentDateMinus30Days.getTime() > Date.now() && creditCardExpirationDate.getTime() > Date.now();
+      }
```

Comment:
Make a function for this

Reply:
I've create a global function to compute the message

Reply:
Push your modifications before resolving the comments please